ad/views.py

```diff
@@ -0,0 +1,23 @@
+from django.shortcuts import render, redirect
+from django.contrib.auth.decorators import login_required
+from django.contrib import messages
+from django.http import HttpResponse
+from .utility import create_post_form, create_get_from
+
+def ad_list(request):
+    return HttpResponse("ad list")
+
+@login_required
+def create_ad(request,adtype):
```

Comment:
Please use generic class based view. You can use django CreateView.

To decide which form to use, you can override get_form_class() method of CreateView

Reply:
Added generic class based view